ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -64,21 +67,29 @@
 
   private final Set<Hash> requestedBlocks = new ConcurrentSet<>();
   private final PendingBlocks pendingBlocks;
+  private final OperationTimer announcedBlockIngestTimer;
 
   BlockPropagationManager(
       final SynchronizerConfiguration config,
       final ProtocolSchedule<C> protocolSchedule,
       final ProtocolContext<C> protocolContext,
       final EthContext ethContext,
       final SyncState syncState,
-      final PendingBlocks pendingBlocks) {
+      final PendingBlocks pendingBlocks,
+      final MetricsSystem metricsSystem) {
     this.config = config;
     this.protocolSchedule = protocolSchedule;
     this.protocolContext = protocolContext;
     this.ethContext = ethContext;
 
     this.syncState = syncState;
     this.pendingBlocks = pendingBlocks;
+
+    this.announcedBlockIngestTimer =
+        metricsSystem.createTimer(
+            MetricCategory.BLOCKCHAIN,
+            "pantheon_blockchain_announcedBlock_ingest",
+            "Time to ingest a single announced block");
```

Comment:
Is it meaningful to track importing an announced block separately to just importing a block?  ie should we just time how long `BlockImporter` takes as that will capture all block imports? I could see it being useful to track how long it takes from first hearing about a block to it being on-chain, but this metric doesn't quite capture that since it only starts the timer after the new block has been retrieved.

Reply:
I was timing the amount of time to process the block.  Its separate code paths IIRc between download and announced blocks, and different tasks (download I see being split into at least three non-adjacent tasks : headers, bodies, receipts).

I could push this down into the persistBlock task, but I was hoping to have the number for the logging statement.

Reply:
Having the timing in the log message is probably justification enough for this.  I think at some point having a timer inside `BlockImporter` itself which will capture every block imported makes sense, but there are going to be a number of different timings related to block import (e.g. timing import of each batch of blocks when synchronizing).

Reply:
Pluming it into tasks is a much larger PR.  I'll leave it here for now, but it may move.